examples/spark2/src/main/scala/org/apache/carbondata/benchmark/ConcurrentQueryBenchmark.scala

```diff
@@ -0,0 +1,631 @@
+    --deploy-mode client \
+    --driver-memory 16g \
+    --executor-cores 4g \
+    --executor-memory 24g \
+    --num-executors 3  \
+    concurrencyTest.jar \
+    totalNum threadNum taskNum resultIsEmpty runInLocal generateFile deleteFile
+ * details in initParameters method of this benchmark
+ */
+object ConcurrentQueryBenchmark {
+
+  // generate number of data
+  var totalNum = 1 * 1000 * 1000
+  // the number of thread pool
+  var threadNum = 16
+  // task number of spark sql query
+  var taskNum = 100
+  // whether is result empty, if true then result is empty
+  var resultIsEmpty = true
+  // the store path of task details
+  var path: String = "/tmp/carbondata"
+  // whether run in local or cluster
+  var runInLocal = true
+  // whether generate new file
+  var generateFile = true
+  // whether delete file
+  var deleteFile = true
+
+  val cardinalityId = 100 * 1000 * 1000
+  val cardinalityCity = 6
+
+  def parquetTableName: String = "Num" + totalNum + "_" + "comparetest_parquet"
+
+  def orcTableName: String = "Num" + totalNum + "_" + "comparetest_orc"
+
+  def carbonTableName(version: String): String =
+    "Num" + totalNum + "_" + s"comparetest_carbonV$version"
+
+  // Table schema:
+  // +-------------+-----------+-------------+-------------+------------+
+  // | Column name | Data type | Cardinality | Column type | Dictionary |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | id          | string    | 100,000,000 | dimension   | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | city        | string    | 6           | dimension   | yes        |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | country     | string    | 6           | dimension   | yes        |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | planet      | string    | 10,007      | dimension   | yes        |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | m1          | short     | NA          | measure     | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | m2          | int       | NA          | measure     | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | m3          | big int   | NA          | measure     | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | m4          | double    | NA          | measure     | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  // | m5          | decimal   | NA          | measure     | no         |
+  // +-------------+-----------+-------------+-------------+------------+
+  /**
+   * generate DataFrame with above table schema
+   *
+   * @param spark SparkSession
+   * @return Dataframe of test data
+   */
+  private def generateDataFrame(spark: SparkSession): DataFrame = {
```

Comment:
It is better to move this to a separate class in benchmark package, it can be reused in two benchmark class

Reply:
ok